src/pretalx/schedule/models/schedule.py

```diff
@@ -157,39 +208,148 @@ def changes(self):
             ).all()
             if talk.is_visible and not talk.submission.is_deleted
         )
+        new_slots_helper = set(
+            (talk.submission, talk.slot_index) for talk in new_slots
```

Comment:
As I said, I'm not a fan of the `slot_index` field – I think counting slots and comparing start time and location should work well enough.

Reply:
done :-)
was a puzzle but i got it working ;-)